apps/emqx_authn/src/emqx_authn_api.erl

```diff
@@ -753,6 +756,140 @@ list_authenticator(ConfKeyPath, AuthenticatorID) ->
             serialize_error(Reason)
     end.
 
+%% example:
+%%
+%% emqx_authn_api:lookup_from_local_node('mqtt:global', <<"password-based:http">>)
+%%
+%% {ok,{'emqx@127.0.0.1',connecting,
+%%      #{counters =>
+%%            #{exception => 0,failed => 0,matched => 0,success => 0},
+%%        rate =>
+%%            #{matched => #{current => 0.0,last5m => 0.0,max => 0.0}}}}}
+lookup_from_local_node(ChainName, AuthenticatorID) ->
+    NodeId = node(self()),
+    case emqx_authentication:lookup_authenticator(ChainName, AuthenticatorID) of
+        {ok, #{provider := Provider, state := State}} ->
+            case lists:member(Provider, resource_provider()) of
+                false -> {error, {NodeId, resource_unsupport_metrics_and_status}};
+                true ->
+                    #{resource_id := ResourceId} = State,
+                    case emqx_resource:get_instance(ResourceId) of
+                        {error, not_found} -> {error, {NodeId, not_found_resource}};
+                        {ok, _, #{ status := Status, metrics := Metrics }} ->
+                            {ok, {NodeId, Status, Metrics}}
+                    end
+            end;
+        {error, Reason} -> {error, {NodeId, Reason}}
+    end.
+
+resource_provider() ->
+    [ emqx_authn_mysql,
+      emqx_authn_pgsql,
+      emqx_authn_mongodb,
+      emqx_authn_redis,
+      emqx_authn_http
+    ].
+
+%% example:
+%%
+%% emqx_authn_api:lookup_from_all_node('mqtt:global', <<"password-based:http">>)
+%%
+%% #{aggregate_metrics =>
+%%       #{counters =>
+%%             #{exception => 0,failed => 0,matched => 0,success => 0},
+%%         rate =>
+%%             #{matched => #{current => 0.0,last5m => 0.0,max => 0.0}}},
+%%   aggregate_status => connecting,all_node_error => #{},
+%%   all_node_metrics =>
+%%       #{'emqx@127.0.0.1' =>
+%%             #{counters =>
+%%                   #{exception => 0,failed => 0,matched => 0,success => 0},
+%%               rate =>
+%%                   #{matched => #{current => 0.0,last5m => 0.0,max => 0.0}}}},
+%%   all_node_state => #{'emqx@127.0.0.1' => connecting}}
+lookup_from_all_node(ChainName, AuthenticatorID) ->
+    Nodes = mria_mnesia:running_nodes(),
+    case is_ok(erpc:multicall(Nodes,
+                              emqx_authn_api,
+                              lookup_from_local_node,
+                              [ChainName, AuthenticatorID],
+                              ?TIMEOUT)) of
+        {ok, ResList} ->
+            {StatusMap, MetricsMap, ErrorMap} = make_result_map(ResList),
+            AggregateStatus = aggregate_status(maps:values(StatusMap)),
+            AggregateMetrics = aggregate_metrics(maps:values(MetricsMap)),
+            #{all_node_state => StatusMap,
+              all_node_metrics => MetricsMap,
+              all_node_error => ErrorMap,
+              aggregate_status => AggregateStatus,
+              aggregate_metrics => AggregateMetrics
```

Comment:
We should make sure the format of `status` and `metrics` returned by this HTTP APIs the same as the one of `GET /bridges`.

```
#{ status => aggregate_status(NodeStatus)
   , node_status => NodeStatus
   , metrics => aggregate_metrics(NodeMetrics)
   , node_metrics => NodeMetrics
   }
```